test/server/api/cache/url.test.ts

```diff
@@ -0,0 +1,59 @@
+import 'reflect-metadata'
+import { spy } from 'sinon'
```

Comment:
Can I ask why not use Jest's [spyOn](https://jestjs.io/docs/en/jest-object.html#jestspyonobject-methodname)?

Reply:
Jest's spyOn doesn't seem to allow matching of only some arguments. In this case, we're only the function being spied on takes in a callback after the parameters and its only necessary to test that the correct parameters are being passed in.

Reply:
just a query - no need to rework if it's too much work.